File: flutter/lib/src/models/native_display_unit.dart
import 'native_display_config.dart';
import 'style.dart';

class NativeDisplayUnit {
  final String unitId;
  final String? slotId;
  final NativeDisplayConfig config;
  final Map<String, dynamic> customExtras;
  final Map<String, Style> resolvedStyles;

  const NativeDisplayUnit({
    required this.unitId,
    this.slotId,
    required this.config,
    this.customExtras = const {},
    this.resolvedStyles = const {},
  });
}

File: react-native/src/bridge/NativeDisplayUnit.ts
import { ResolvedConfig, ResolvedStyles } from '../models/NativeDisplayConfig';

export interface NativeDisplayUnit {
  unitId: string;
  config: ResolvedConfig;
  resolvedStyles: ResolvedStyles;
  slotId?: string;
  customExtras: Record<string, string>;
  rawJson?: string;
}
